apps/framework-cli/src/cli/routines/migrate.rs

```diff
@@ -313,19 +305,66 @@ pub async fn execute_migration(
             e,
         )
     })?;
-    let client = create_client(clickhouse_config.clone());
-    check_ready(&client).await.map_err(|e| {
-        RoutineFailure::new(
-            Message::new(
-                "ClickHouse".to_string(),
-                "Failed to connect to ClickHouse".to_string(),
-            ),
-            e,
-        )
-    })?;
 
-    // Create state storage directly from CLI-provided ClickHouse URL
-    let state_storage = ClickHouseStateStorage::new(client, clickhouse_config.db_name.clone());
+    // Build state storage based on config
+    let state_storage: Box<dyn StateStorage> = match project.state_config.storage.as_str() {
+        "redis" => {
```

Comment:
Feels like this should be in the state storage building.

Why did you pull it out?

Reply:
Good catch, I cleaned things up a bit